tests/generic_config_updater/test_cacl.py

```diff
@@ -2,110 +2,151 @@
 import pytest
 
 from tests.common.helpers.assertions import pytest_assert
-from tests.common.config_reload import config_reload
 from tests.generic_config_updater.gu_utils import apply_patch, expect_op_success, expect_res_success, expect_op_failure
 from tests.generic_config_updater.gu_utils import generate_tmpfile, delete_tmpfile
+from tests.generic_config_updater.gu_utils import create_checkpoint, delete_checkpoint, rollback_or_reload
+
+# Test on t0 topo to verify functionality and to choose predefined variable
+# admin@vlab-01:~$ show acl table
+# Name        Type       Binding          Description    Stage
+# ----------  ---------  ---------------  -------------  -------
+# ...
+# NTP_ACL     CTRLPLANE  NTP              NTP_ACL        ingress
+# SNMP_ACL    CTRLPLANE  SNMP             SNMP_ACL       ingress
+# SSH_ONLY    CTRLPLANE  SSH              SSH_ONLY       ingress
 
 pytestmark = [
-    pytest.mark.topology('any'),
+    pytest.mark.topology('t0'),
 ]
 
 logger = logging.getLogger(__name__)
 
-@pytest.fixture(scope="module", autouse=True)
-def setup_env(duthosts, rand_one_dut_hostname, cfg_facts):
+T0_CACL_TABLE = ["NTP_ACL", "SNMP_ACL", "SSH_ONLY"]
+
+def get_cacl_tables(duthost):
+    """Get acl control palne tables
+    """
+    cmds = "show acl table | grep -w CTRLPLANE | awk {'print $1'} || true"
```

Comment:
> || true [](http://example.com/codeflow?start=65&length=8)

Not to ignore return code. If anything other than grep match failed, we need to capture and report an error.

Reply:
Removed `true` to ignore errors.
Have some test on below scenario, it will not report an error because return code is still `0`.
The only difference matters is whether the last command succeed or not.
```
admin@vlab-01:~$ showwwww acl table | grep -w CTRLPLANE | awk {'print $1'}
-bash: showwwww: command not found
admin@vlab-01:~$ echo $?
0
admin@vlab-01:~$ show acl table | greeeeep -w CTRLPLANE | awk {'print $1'}
-bash: greeeeep: command not found
admin@vlab-01:~$ echo $?
0
admin@vlab-01:~$ show acl table | grep -w CTRLPLANE | aaaaawk {'print $1'}
-bash: aaaaawk: command not found
admin@vlab-01:~$ echo $?
127
```
Also tested in sonic-mgmt, the return code will not be `0` only when the last pipe command fails. So it will not report error even first several commands fail.
```
{'stderr_lines': [u'/bin/sh: 1: shows: not found'], u'cmd': u"shows acl table | grep -w CTRLPLANE | awk {'print $1'}", u'end': u'2021-12-24 04:23:40.000951', '_ansible_no_log': False, u'stdout': u'', u'changed': True, u'rc': 0, u'start': u'2021-12-24 04:23:39.995670', u'stderr': u'/bin/sh: 1: shows: not found', u'delta': u'0:00:00.005281', u'invocation': {u'module_args': {u'creates': None, u'executable': None, u'_uses_shell': True, u'strip_empty_ends': True, u'_raw_params': u"shows acl table | grep -w CTRLPLANE | awk {'print $1'}", u'removes': None, u'argv': None, u'warn': True, u'chdir': None, u'stdin_add_newline': True, u'stdin': None}}, 'stdout_lines': [], 'failed': False}
```
Maybe the `shell` command should report an error based on non-empty `stderr` instead of non-zero `rc` ?

Reply:
Thanks for the experiment details! LGTM now.

If you would like to investigate further, you may check `set -e` and `set -o pipefail`, ref: https://stackoverflow.com/a/4959616/2514803.

Or consider use python to invoke individual commands, check return codes, and chain their stdout/stdin together. This is difficult since it is invoke on duthost.